Core/VariantManager.swift

```diff
@@ -25,34 +25,49 @@ public enum FeatureName: String {
     // Used for unit tests
     case dummy
     
+    case iPadImprovements
     case tabSwitcherListLayout
     
 }
 
-public enum CohortFiltering {
+public enum VariantInclusion {
     
-    case includeInCohort
-    case excludeFromCohort
+    case always
+    case when(_ filter: () -> Bool)
     
 }
 
 public struct Variant {
     
+    struct Is {
+        static let padDevice = { return UIDevice.current.userInterfaceIdiom == .pad }
+        static let notPadDevice = { return !Is.padDevice() }
+    }
+    
     static let doNotAllocate = 0
     
+    // Note: Variants with `doNotAllocate` weight, should always be included so that previous installations are unaffected
```

Comment:
Can we add a note that all variants for an experiment must have the same filters? It would be invalid for example to have no filter on the control group and them make the experimental group English-only.